src/coreclr/pal/src/map/virtual.cpp

```diff
@@ -2185,8 +2197,49 @@ void ExecutableMemoryAllocator::TryReserveInitialMemory()
 
     } while (sizeOfAllocation >= MemoryProbingIncrement);
 
+#else // TARGET_XARCH
+
+    if ((coreclrLoadAddress < 0xFFFFFFFF) || ((coreclrLoadAddress - MaxExecutableMemorySizeNearCoreClr) < 0xFFFFFFFF))
```

Comment:
Can we keep this code common instead of splitting it per architecture?

Reply:
> Can we keep this code common instead of splitting it per architecture?

Addressed